base/strings/types.jl

```diff
@@ -122,6 +124,8 @@ reverse(s::RevString) = s.string
 
 isascii(s::RevString{ASCIIString}) = true
 
+Base.convert{T<:AbstractString}(::Type{RevString{T}}, x::SubString{RevString{T}}) = RevString(reverse(bytestring(x)))
```

Comment:
Unfortunately I get some recursion/stack overflow if I do that, between the first of the convert methods I add and this method. I don't know how to fix unfortunately...